docs/src/main/sphinx/security/user-mapping.rst

```diff
@@ -2,67 +2,123 @@
 User mapping
 ============
 
-User mapping defines rules for mapping from users in the authentication system to Trino users.  This
-mapping is particularly important for Kerberos or certificate authentication where the user names
-are complex like ``alice@example`` or ``CN=Alice Smith, OU=Finance, O=Acme, C=US``.
+User mapping defines rules for mapping from users in the authentication method to Trino users. This
+mapping is particularly important for :doc:`Kerberos </security/kerberos>` or
+certificate authentication where the user names
+are complex, such as ``alice@example`` or ``CN=Alice Smith, OU=Finance, O=Acme, C=US``.
 
-User mapping can be configured with a simple regex extraction pattern, or more complex rules in a
-separate configuration file.
+There are two ways to map the username format of a given authentication
+provider into the simple username format of Trino users:
+
+* With a single regular expression (regex) :ref:`pattern mapping rule <pattern-rule>`
+* With a :ref:`file of regex mapping rules <pattern-file>` in JSON format
+
+.. _pattern-rule:
 
 Pattern mapping rule
 --------------------
 
-The pattern mapping rule maps the authentication user to the first matching group in the regular
-expression.  If the regular expression does not match the authentication user, authentication is
-denied.
+If you can map all of your authentication method’s usernames with a single
+reqular expression, consider using a **Pattern mapping rule**.
+
+For example, your authentication method might have all usernames in the form
+``alice@example.com``, with no exceptions. In this case, choose a regex that
+breaks incoming usernames into at least two regex capture groups, such that the
+first capture group includes only the name before the @ sign. You can use the
+simple regex ``(.*)(@.*)`` for this case.
+
+Trino automatically uses the first capture group – the $1 group – as the
+username to emit after the regex substitution. If the regular expression does
+not match the incoming username, authentication is denied.
+
+Specify your regex pattern in a property in your coordinator’s
+``config.properties`` file, using the ``...user-mapping.pattern`` property from
+the table below that matches the authentication type of your configured
+authentication provider. For example, for an :doc:`LDAP </security/ldap>`
+authentication provider:
 
-Each authentication system has a separate property for the user mapping pattern to allow different mapping
-when multiple authentication systems are enabled:
+.. code-block:: text
+
+    http-server.authentication.password.user-mapping.pattern="(.*)(@.*)"
+
+Remember that an :doc:`authentication type </security/authentication-types>`
+represents a category, such as PASSWORD, OAUTH2, KERBEROS. More than one
```

Comment:
Should wrap these authentication types in code font

Reply:
Correct, with thanks.